### ISSUE TITLE
unreadable_literal triggers for valid scientific notation

### ISSUE THREAD

**Issue by killercup** [rust-lang-nursery/rust-clippy#2487]
This looks like a false positive:

```
warning: long literal lacking separators
  --> src/consts.rs:10:20
   |
10 | pub const G: f64 = 6.67e-11_f64;
   |                    ^^^^^^^^^^^^
   |
   = note: #[warn(unreadable_literal)] on by default
   = help: consider: 6.67e_-11_f64
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.186/index.html#unreadable_literal
```

**Comment by clarcharr** [comment]
Yeah, afaik the lint just checks the length of the string and not its contents…
